test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
 )
 
 
+from kornia.testing import BaseTester
+from kornia.augmentation.base import AugmentationBase2D
+
+# TOTEST
+# _validate_input_dtype
+# _transform_input
+# identity_matrix
+
+# Common
+# Given arbitrary input shapes the output dims are consistent
+# shape changing not shape changing
+#  return_transform: bool = False, same_on_batch: bool = False, p: float = 0.5,
+#  p_batch: float = 1.
+# Smoke tests, lehet példányosítani, meg lehet hívni az abstract függvényeit stb
+# generate_parameters annyi paramot ad vissza amennyi a batch size
+# compute_transformation implementálva van és batchsiye stimmel
+# apply_transform implementálva van és batchsiye stimmel
+
+
+class CommonTests(BaseTester):
```

Comment:
It's only my personal preference, but I think too many levels of abstraction and hence deep hierarchy makes the code sometimes really hard to understand. I'd go for something like CommonTests2D and CommonTests3D etc. It's not easy to define an absolut shared set of tests because a lot of some exceptions, but I think the cleanest would be in those cases to just override the necessary test cases or skip the ones that doesn't apply.

Reply:
The only difference here is that CommonTests2D is doing (BCHW) while CommonTests3D is doing (BCDHW). What I am thinking is to have a CommonTester that can perform 1) shape-agnostic testings like test_smoke, and 2) some abstracted testings without shape awareness like "output.shape == expected_output.shape".

Then the child classes will do things like 1) shape-specific testings like testing HW, CHW, BCHW for 2D while DHW, CDHW, BCDHW for 3D, and 2) dimension-specific testing like the 3x3 return matrix for 2D and 4x4 return matrix for 3D.

Reply:
I would rename this class as `CommonAugmentationTests` since it's only consumed by the augmentation module operators.